src/metabase/api/collection.clj

```diff
@@ -447,6 +454,18 @@
   {namespace (s/maybe su/NonBlankString)}
   (dissoc (root-collection namespace) ::collection.root/is-root?))
 
+(defn- visible-model-kwds
+  "If you pass in explicitly keywords that you can't see, you can't see them.
+  But there is an exception for the collections,
+  because you might not be able to see the top-level collections
+  but be able to see, children of those invisible top-level collections."
+  [root-collection model-set]
+  (if (mi/can-read? root-collection)
+    model-set
+    (if (or (empty? model-set) (contains? model-set :collection))
+      #{:collection}
+      #{:dummy})))
```

Comment:
Wouldn't `:dummy` be the model that `collection-children-query` dispatches on and that would throw a 
> No method in multimethod 'collection-children-query' for dispatch value: :dummy

Reply:
dummy goes through the kwd for-loop and gets filtered out by the when, and turns into a nil set and nothing gets dispatched. if you pass in a nil set you get set of all models...